node/network/protocol/src/request_response/request.rs

```diff
@@ -284,6 +281,98 @@ where
 	}
 }
 
+/// Typed variant of [`netconfig::OutgoingResponse`].
+///
+/// Responses to `IncomingRequest`s.
+pub struct OutgoingResponse<Response> {
+	/// The payload of the response.
+	pub result: Result<Response, ()>,
```

Comment:
I am curious about the use of `()` as the `Err` type - should it not be more expressive?

Reply:
That's just mimicking the more low level substrate API - copied the comment from there for clarification.